Dfe.Data.SearchPrototype/Web/Models/ViewModels/Shared/ScrollablePager.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Dfe.Data.SearchPrototype.Web.Models.ViewModels.Shared
+{
+    /// <summary>
+    /// Provides functionality to allow paging to work in a scrollable manner whereby moving forwards or
+    /// backwards across a given page sequence means the page selected is maintained in the middle of the sequence,
+    /// thus giving the impression that the pages are scrolling forwards or backwards depending on the selection made.
+    /// </summary>
+    public sealed class ScrollablePager : IPager
+    {
+        /// <summary>
+        /// Sets the default number of lower and upper pages to 2. This means that
+        /// if the current page falls within this lower or upper page band, certain
+        /// constraints can be triggered (e.g. display/hide next previous buttons).
+        /// </summary>
+        private const int PageSequencePaddingSize = 2;
+
+        /// <summary>
+        /// Used to determine if FirstPageNumber should be displayed ahead of PageSequence
```

Comment:
i.e. returns true if the PageSequence includes page 1
method name could be PageSequenceIncludesFirstPage maybe?